Archipelago.HollowKnight/Archipelago.cs

```diff
@@ -167,6 +247,12 @@ private void ConnectToArchipelago()
                 NotchCosts = SlotDataExtract.ExtractArrayFromSlotData<List<int>>(success.SlotData["notch_costs"]);
                 SlotOptions = SlotDataExtract.ExtractObjectFromSlotData<SlotOptions>(success.SlotData["options"]);
 
+                if(SlotOptions.Goal < 0 || SlotOptions.Goal > Goal.MAX)
+                {
+                    // TODO: Should probably yell about using an outdated client here.  I'm not certain the best way to do that, so just default it to ANY for now
+                    // --Dewin
+                    SlotOptions.Goal = Goal.Any;
```

Comment:
Throw an exception here, we can bubble up the exception message to the user in ArchipelagoModeMenuConstructor.cs#L76 instead of defaulting it to tell people to contact me.